### ISSUE TITLE
Downloads over http/2 freeze if max-jobs is set to > 1

### ISSUE THREAD

**Issue by glittershark** [NixOS/nix#3438]
If I `nix-build` a derivation that downloads something from cache.nixos.org with `max-jobs` set to `auto` (in my `nix.conf`), the downloads freeze at 0% and fail to progress. Passing `--max-jobs 1` fixes the issue, as does `--option http2 false`, and the issue is not present on my own binary cache which doesn't support http/2.

Not clear enough with what's going on here to provide more diagnostic information, but happy to provide anything that might help.

**Comment by glittershark** [comment]
@andir 
```
❯ yay -Qs libcurl
local/curl 7.69.0-1
    An URL retrieval utility and library
local/libcurl-compat 7.69.0-1
    An URL retrieval library (without versioned symbols)
local/libcurl-gnutls 7.69.0-1
    An URL retrieval library (linked against gnutls)
```

**Comment by andir** [comment]
This might be related to https://github.com/NixOS/nix/issues/2733

**Comment by andir** [comment]
On 12:30 22.03.20, Griffin Smith wrote:
> @andir 
> ```
> ❯ yay -Qs libcurl
> local/curl 7.69.0-1
>     An URL retrieval utility and library
> local/libcurl-compat 7.69.0-1
>     An URL retrieval library (without versioned symbols)
> local/libcurl-gnutls 7.69.0-1
>     An URL retrieval library (linked against gnutls)
> ```

What libcurl does your nix actually link against? Can you post the
output of `ldd` on it?

Can you try using a nix that was built with nix (e.g. from nixpkgs master)?

**Comment by glittershark** [comment]
```

**Comment by glittershark** [comment]
> Can you try using a nix that was built with nix (e.g. from nixpkgs master)?
nix commands inside of `nix-shell -p nix` fail with 
```
nix: error while loading shared libraries: libsqlite3.so.0: cannot open shared object file: No such file or directory
```
which I think is a separate error, maybe related to `archlinux-nix` stuff. I'm not super familiar with how any of this works

Oh, yeah, I should mention this is on Arch, with `archlinux-nix`

**Comment by edolstra** [comment]
This may be a dupe of #3399. If so it would be fixed by curl 7.69.1.

**Comment by glittershark** [comment]
Yep, curl 7.69.1 fixes it. Thanks!
